src/cloud/internal/cloudservice.h

```diff
@@ -55,7 +57,10 @@ class CloudService : public QObject, public IAuthorizationService, public IUploa
     ValCh<bool> userAuthorized() const override;
     ValCh<AccountInfo> accountInfo() const override;
 
-    framework::ProgressChannel uploadScore(const QByteArray& scoreData, const std::string& title, const QUrl& sourceUrl = QUrl()) override;
+    void uploadScore(system::IODevice& scoreSourceDevice, const QString& title, const QUrl& sourceUrl = QUrl()) override;
+
+    async::Channel<QUrl> sourceUrlReceived() const override;
+    framework::ProgressChannel progressChannel() const override;
```

Comment:
It's strange that `ProgressChannel` in the namespace is a framework, not an namespace async  :)

Reply:
Progress is stored in mu::global. ProgressChannel just an alias. We can put ProgressChannel into mu::async, but it seems a little strange for me